adbc.h

```diff
@@ -459,6 +459,24 @@ const struct AdbcError* AdbcErrorFromArrayStream(struct ArrowArrayStream* stream
 ///
 /// \see AdbcConnectionGetInfo
 #define ADBC_INFO_VENDOR_ARROW_VERSION 2
+/// \brief Indicates whether SQL queries are supported (type: bool).
```

Comment:
Should we also make a new value for `ADBC_VERSION` ? `1.2.0` for example?

Reply:
There's some discussion about this in #1650. What do you think about version `1.1.1`?

Reply:
Do you mean for the version returned in GetInfo, or also making a new constant that would be recognized by the driver manager and such?

Reply:
As I'm looking through the code now for references to the constant `ADBC_VERSION_1_1_0`, it does seem like a non-trivial amount of work to update the logic everywhere to account for `ADBC_VERSION_1_1_1`. Especially for such a minor and backward-compatible change, it doesn't seem worth it.

Perhaps we could just do a comment / documentation update here to denote the minor version bump? Also, a refactor of ADBC version handling might be helpful going forward if it would simplify the logic for various feature gates that would allow us to bump versions more seamlessly.

Reply:
We basically treat the version as an ABI version. Adding constants doesn't change the API, so there's no need to bump.

Reply:
Or at least, that's my justification for not having to deal with that mess here ^_^

Reply:
I think considering the ADBC version to be an ABI version makes sense. If a driver implementation were to adopt this "version" it would not require any changes in behavior, as there's no actual requirement to send or recognize the new constants.

I think it may actually be more confusing to describe this as a new version, since there are no discernible differences in behavior between 1.1.0 and 1.1.1.

Any objections to moving forward with this PR as it currently is?

Reply:
I'm happy with that.

Reply:
Sounds good to me too -- sorry for not responding sooner. Thank  you @joellubi  and @lidavidm